Assets/Scripts/EnemyController.cs

```diff
@@ -124,6 +124,10 @@ void FindNavTarget()
                     }
                 }
             }
+            if (closestPlayer == null)
+            {
+                return;
+            }
             player = closestPlayer;
             agent.destination = closestPlayer.position;
             refreshTargetTimer = refreshTargetTimerLimit;
```

Comment:
you could've done this whole part before that a bit simpler, cause if there are no alive players, no closest player will be found. But I'm just giving this as very quick code improvement
```suggestion
            if (closestPlayer != null)
            {
                   player = closestPlayer;
                   agent.destination = closestPlayer.position;
                   refreshTargetTimer = refreshTargetTimerLimit;
            }
```